signer/src/bitcoin/client.rs

```diff
@@ -0,0 +1,84 @@
+//! Bitcoin Core RPC client implementations
+
+use bitcoincore_rpc::RpcApi;
+use sbtc::rpc::{BitcoinClient, BitcoinCoreClient};
+use url::Url;
+
+use crate::{error::Error, keys::PublicKey, util::ApiFallbackClient};
+
+use super::{
+    utxo::{self, SignerUtxo},
+    BitcoinInteract,
+};
+
+/// Implement the [`TryFrom`] trait for a slice of [`Url`]s to allow for a
+/// [`ApiFallbackClient`] to be implicitly created from a list of URLs.
+impl TryFrom<&[Url]> for ApiFallbackClient<BitcoinCoreClient> {
+    type Error = Error;
+    fn try_from(urls: &[Url]) -> Result<Self, Self::Error> {
+        let clients = urls
+            .iter()
+            .map(|url| BitcoinCoreClient::try_from(url.clone()).unwrap())
```

Comment:
Fixed in [7574dd2](https://github.com/stacks-network/sbtc/pull/549/commits/7574dd2a9d48a462426cf8874308485add831a29)

Note: I added the following to catch this kind of thing pre-PR. I added `unwrap_used` and `expect_used` but commented out since there's some left I don't want to address them in this PR, but I think it feels like a pretty good goal to shoot for being able to enable these. Thoughts? They're disabled again in the `testing` mod.

```rust
#![deny(
    clippy::unwrap_in_result,
    // clippy::unwrap_used, // TODO: There's one unwrap left
    // clippy::expect_used, // TODO: There's 14 expects left
)]
```

Reply:
I think `clippy::unwrap_in_result` is a good one to add, no objection from me there. And `clippy::unwrap_used` seems like a good one as well. Outside of tests it is always preferable to use `expect` instead of `unwrap` since you are forced to give context. But with `expect`, there are times where its use is perfectly fine. For example, when "writing" to a `Vec` when using the `std::io::Write` trait, using `expect` is fine. Maybe the computer runs out of memory but the `std::io::Write::write` call won't return a `Result::Err`. So I feel like `.expect` should be viewed with suspicion but not strictly forbidden.

Reply:
I kinda figured that in the cases where you want to explicitly use any of the above that one would also explicitly disable the clippy warning with a comment on why it's OK to use.

Reply:
> I kinda figured that in the cases where you want to explicitly use any of the above that one would also explicitly disable the clippy warning with a comment on why it's OK to use there. That makes it clear that it's no accident.

👍🏽 oh yeah, that is better.